analytics/javaspark/src/main/java/mil/nga/giat/geowave/analytic/javaspark/kmeans/KMeansRunner.java

```diff
@@ -102,14 +107,50 @@ public void run()
 		queryOptions.setAdapter(featureAdapterIds);
 
 		// This is required due to some funkiness in GeoWaveInputFormat
-		queryOptions.getAdaptersArray(inputDataStore.createAdapterStore());
+		AdapterStore adapterStore = inputDataStore.createAdapterStore();
+		queryOptions.getAdaptersArray(adapterStore);
 
 		// Add a spatial filter if requested
 		DistributableQuery query = null;
-		if (bbox != null) {
-			Geometry geom = new GeometryFactory().toGeometry(bbox);
-			query = new SpatialQuery(
-					geom);
+		try {
+			if (cqlFilter != null) {
+				Geometry bbox = null;
+				ByteArrayId cqlAdapterId;
+				if (adapterId == null) {
+					cqlAdapterId = featureAdapterIds.get(0);
+				}
+				else {
+					cqlAdapterId = new ByteArrayId(
+							adapterId);
+				}
+
+				DataAdapter adapter = adapterStore.getAdapter(cqlAdapterId);
+
+				if (adapter instanceof FeatureDataAdapter) {
+					final String geometryAttribute = ((FeatureDataAdapter) adapter)
+							.getFeatureType()
+							.getGeometryDescriptor()
+							.getLocalName();
+					Filter filter;
```

Comment:
odd, must have been an artifact of something OBE for these two lines to be separate. iNot worth disapproving the PR just for that.